client/src/hooks/reducers/feedbackReducers.js

```diff
@@ -22,7 +22,13 @@ export const initialState = {
 };
 
 export const feedbackReducer = (oldState, action) => {
-  const { type, key, value } = action;
+
+  // this is only for type number inputs
+  if (action.limit && action.value < action.limit.min) action.value = action.limit.min;
+  if (action.limit && action.value > action.limit.max) action.value = action.limit.max;
+
```

Comment:
Have you tried just using vanilla `max` and `maxlength` attributes for inputs without any of this? and adding type attributes for text/number inputs, or there was any issues with using that alone?

Reply:
Yes I tried, but there's a problem with the `max` attribute as it does not prevent users from typing any number, and the form is still being submitted because the submit button is in fact just an anchor tag. HTML would prevent it if it was a type submit button wrapped in an actual form element but that's not the case and given the way everything is abstracted in the code, I wouldn't know how to make such change without breaking stuff xD

Reply:
Yeah then probably just "substring" numbers lol (sounds weird), so they don't turn into 999.